Assets/MixedRealityToolkit.Examples/Experimental/SceneUnderstanding/Scripts/DemoSceneUnderstandingController.cs

```diff
@@ -0,0 +1,395 @@
+
+            CoreServices.SpatialAwarenessSystem.RegisterHandler<IMixedRealitySpatialAwarenessObservationHandler<SpatialAwarenessSceneObject>>(this);
+
+            var access = CoreServices.SpatialAwarenessSystem as IMixedRealityDataProviderAccess;
+
+            if (access == null)
+            {
+                Debug.LogError("Couldn't get access to CoreServices.SpatialAwarenessSystem");
+                return;
+            }
+
+            observer = access.GetDataProvider<IMixedRealitySpatialAwarenessSceneUnderstandingObserver>();
+
+            if (observer == null)
+            {
+                Debug.LogError("Couldn't access Scene Understanding Observer!");
+                return;
+            }
+        }
+
+        async void OnEnable()
+        {
+            await RegisterHandlersAsync();
+            platforms.Clear();
+        }
+
+        void OnDisable()
+        {
+            CoreServices.SpatialAwarenessSystem?.UnregisterHandler<IMixedRealitySpatialAwarenessObservationHandler<SpatialAwarenessSceneObject>>(this);
+        }
+
+        async void Update()
+        {
+            if (!isInit && (observer != null))
+            {
+                InitToggleButtonState();
+                isInit = true;
+            }
+
+            if (autoPlaceStuffOnce)
+            {
+                autoPlaceStuffOnce = false;
+                await Task.Delay(TimeSpan.FromSeconds(4));
+                PutStuffOnNearestPlatform();
+            }
+        }
+
+        private void InitToggleButtonState()
+        {
+            autoUpdateToggle.GetComponent<Interactable>().IsToggled = observer.AutoUpdate;
+            quadsToggle.GetComponent<Interactable>().IsToggled = observer.RequestPlaneData;
+            meshesToggle.GetComponent<Interactable>().IsToggled = observer.RequestMeshData;
+            maskToggle.GetComponent<Interactable>().IsToggled = observer.RequestOcclusionMask;
+            platformToggle.GetComponent<Interactable>().IsToggled = observer.SurfaceTypes.HasFlag(SpatialAwarenessSurfaceTypes.Platform);
+            wallToggle.GetComponent<Interactable>().IsToggled = observer.SurfaceTypes.HasFlag(SpatialAwarenessSurfaceTypes.Wall);
+            floorToggle.GetComponent<Interactable>().IsToggled = observer.SurfaceTypes.HasFlag(SpatialAwarenessSurfaceTypes.Floor);
+            ceilingToggle.GetComponent<Interactable>().IsToggled = observer.SurfaceTypes.HasFlag(SpatialAwarenessSurfaceTypes.Ceiling);
+            worldToggle.GetComponent<Interactable>().IsToggled = observer.SurfaceTypes.HasFlag(SpatialAwarenessSurfaceTypes.World);
+            completelyToggle.GetComponent<Interactable>().IsToggled = observer.SurfaceTypes.HasFlag(SpatialAwarenessSurfaceTypes.CompletelyInferred);
+            backgroundToggle.GetComponent<Interactable>().IsToggled = observer.SurfaceTypes.HasFlag(SpatialAwarenessSurfaceTypes.Background);
+            usePersistentToggle.GetComponent<Interactable>().IsToggled = observer.UsePersistentObjects;
```

Comment:
Consider making each of the GameObject references, just Interactable references instead?